docs/dynamic-programming/cutting-a-rod.mdx

```diff
@@ -0,0 +1,178 @@
+---
+title: Cutting a rod
+slug: /dynamic-programming/cutting-a-rod
+---
+
+## Introduction
+
+Given a rod of length `n` and an array of prices of all pieces of length `1` to `n`, we need to determine the maximum price that can be obtained by cutting up the rod and selling its pieces.
```

Comment:
The introduction is not very intuitive. A better explanation would be needed here.